contract/contracts/reverse_name_resolver/reverse_name_resolver.ral

```diff
@@ -0,0 +1,39 @@
+Contract ReverseNameResolver() extends Constants(), Events() {
+
+    mapping[Address, ByteVec] addressNames
+
+    pub fn getNameByAddress(address: Address) -> ByteVec {
+        return addressNames[address]
+    }
+
+    pub fn containsNameByAddress(address: Address) -> Bool {
+        return addressNames.contains!(address)
+    }
+
+    @using(preapprovedAssets = true, checkExternalCaller = false)
+    pub fn setAddressName(name: ByteVec) -> () {
+        let caller = callerAddress!()
+        if (addressNames.contains!(caller)) {
+            addressNames.remove!(caller, caller)
```

Comment:
If the map entry exists, we can update it to save some gas

Reply:
done